tools/dev-doctor: remove vagrant dev VM specific checks

tklauser opened the pull request:
The NFS config and iptables rules checks in dev-doctor were only relevant for the deprecated vagrant dev VM setup which was removed in commit bafb49b7379f ("cleanup: Remove vagrant related file").

aditighag reviewed (approved):
Thanks! 🧹 GHA is highlighting an unused function.

sayboras reviewed (approved):
Ship it :)

tklauser committed:
tools/dev-doctor: remove vagrant dev VM specific checks

The NFS config and iptables rules checks in dev-doctor were only
relevant for the deprecated vagrant dev VM setup which was removed in
commit bafb49b7379f ("cleanup: Remove vagrant related file").

Signed-off-by: Tobias Klauser <tobias@cilium.io>
